src/Blockfrost.Api/Extensions/BlockfrostServiceExtensions.cs

```diff
@@ -122,8 +122,8 @@ public static IServiceCollection AddBlockfrost(this IServiceCollection services,
             if (connectionLimit > Constants.CONNECTION_LIMIT){
                 throw new ArgumentOutOfRangeException(nameof(connectionLimit), $"Connections are limited to {Constants.CONNECTION_LIMIT}");
             }
-            services.TryAddScoped(_ => new BlockfrostAuthorizationHandler(apiKey));
-            services.TryAddScoped(_ => new RateLimitHandler());
+            services.TryAddSingleton(_ => new BlockfrostAuthorizationHandler(apiKey));
+            services.TryAddSingleton(_ => new RequestThrottler());
```

Comment:
Why not? We can use the same AuthHandler as we will have the same API key for every service. And we wand to use the same throttling handler for all services because we have a rate limit on API level, not endpoint level (as far as I understood)